File: typescript/session2/heritageMultiple/SuccDecimal.ts
import {EtatSucc} from "./EtatSucc";
import {AlgebreNatDecimal} from "./AlgebreNatDecimal";
import {Nat} from "../../session1/td/Nat/Nat";
import {ZeroDecimal} from "./ZeroDecimal";

export class SuccDecimal extends EtatSucc, AlgebreNatDecimal {

    constructor(rep : Nat) {
        super(rep);
    }

    creerNatZero() : Nat {
        return new ZeroDecimal();
    }

    creerNatSucc(predecesseur : Nat) : Nat{
        return new SuccDecimal(predecesseur);
    }

    toString() : string {
        return '' + this.val();
    }

    equals( obj : Object) : boolean{
        if (!(obj instanceof Nat))
            return false;
        let x = <Nat> obj;
        return this.val() == x.getVal();
    }
}

File: java/cours/src/session2/td/heritageMultiple/SuccDecimal.java
package session2.td.heritageMultiple;

import session1.td.Nat;

public class SuccDecimal extends EtatSucc implements AlgebreNatDecimal {

	public SuccDecimal(Nat rep) {
		super(rep);
	}
	
	@Override
	public Nat creerZero() {
		return new ZeroDecimal();
	}
	
	@Override
	public Nat creerSuccesseur(Nat predecesseur) {
		return new SuccDecimal(predecesseur);
	}
	
	@Override
	public String toString() {
		return Integer.toString(this.val());
	}

	@Override
	public boolean equals(Object obj) {
		if (!(obj instanceof Nat))
			return false;
		Nat x = (Nat) obj;
		return this.val() == x.val();
	}
}
